### PR TITLE
chore: migrate bootstrap.sh to main branch

### PR DESCRIPTION
**Pull request by meltsufin**
This PR migrates `bootstrap.sh` and `repos.txt` from `monorepo_script` to `main`. This allows us to regenerate the monorepo and create diffs in `generation/monorepo` without needing the `monorepo_script` branch. This also allows us to continuously test most of the scripts in `generation/` via CI executions of `bootstrap.sh`.

### CODE REVIEW

File: generation/bootstrap.sh [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+git add google-cloud-gapic-bom/pom.xml
+git commit -am 'feat: create google-cloud-gapic-bom'
+
+cp ../../gapic_bom_versions.txt google-cloud-gapic-bom/versions.txt
+
+# Generate Release Please configuration files
+bash ../../generate_release_please_config.sh
+
+git add --all
+git commit -am 'feat: create release please configuration'
```
Comment (ddixit14) [inline generation/bootstrap.sh:97]: Can we change the commit prefix to `chore` ? Otherwise it gets picked up in release-notes.

Comment (meltsufin) [reply generation/bootstrap.sh]: Done.

**Review by ddixit14 — APPROVED** [review]

File: .github/workflows/bootstrap.yaml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        run: mvn compile -T C1 -B
+      - name: Unit Tests
+        working-directory: generation/monorepo/google-cloud-java
+        run: mvn test -T C1 -B
+      - name: Push monorepo to branch
+        if: ${{ github.event_name != 'pull_request' }}
+        run: |
+          cd generation/monorepo/google-cloud-java
+          git remote add monorepo https://${{ github.actor }}:${{ github.token }}@github.com/${{ github.repository }}.git
+          git push -f monorepo main:bootstrap_output
```
Comment (suztomo) [inline .github/workflows/bootstrap.yaml:36]: Isn't the target branch name monorepo_script_output?

File: .github/workflows/bootstrap.yaml [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+      - name: Push diff to branch
+        run: |
+          cd generation/monorepo/diff
+          git remote add monorepo https://${{ github.actor }}:${{ github.token }}@github.com/${{ github.repository }}.git
+          git push -f monorepo main:boootstrap_output_diff_${{ github.event_name }}
```
Comment (suztomo) [inline .github/workflows/bootstrap.yaml:41]: Isn't the target branch name monorepo_script_output?

File: .github/workflows/bootstrap.yaml (hunk 2 above)
Comment (meltsufin) [reply .github/workflows/bootstrap.yaml:36]: I'm changing it, to reduce confusion.

File: .github/workflows/bootstrap.yaml (hunk 3 above)
Comment (meltsufin) [reply .github/workflows/bootstrap.yaml:41]: This is the output for the diff part of the script.
Notice `cd generation/monorepo/diff` above.

**Review by suztomo — APPROVED** [review]

**Pull request merged**